### ISSUE TITLE
Not catching any pokemons

### ISSUE THREAD

**Issue by sffetlio** [PokemonGoF/PokemonGo-Bot#821]
Since I updated to dev, the bot is not catching any pokemons.
The log shows 20+ pokestops and no pokemons between them.

"mode": "all" in the config

### Other Information
OS: Ubuntu
fa9df3b05072432f3ca1f9d68a39cf086a4686eb

**Comment by alvarogzp** [comment]
same here, only catches pokemons at the beginning, then it starts going to pokestops and forgets about pokemons

**Comment by LeaklessGfy** [comment]
This is a critical bug that we are trying to fix at the moment ...
Not sure it will be fix in the next hours.

If you got a commit version, branch or tagged that work please refer the name here, like that we can try to move quickly.

Thanks :)

**Comment by sffetlio** [comment]
I have 6e99169d963bd8637dfb55426650f2fe15e694bd and it's working there.

**Comment by tstumm** [comment]
Does the problem still persist when using "poke" as farming mode?

**Comment by gpires** [comment]
@tstumm: "poke" works, but "all" doesn't.

Commit: 32cefeed4760d6b5d2b3eb76d783d0f13d068e83

**Comment by z4ppy** [comment]
Same here, last version (33a0ad6) and no capturing pokemon with "all".

**Comment by josephmcox** [comment]
Even in mode poke not much is being caught. Could this be an issue with the new google directions commit?

**Comment by josephmcox** [comment]
Confirmed if i checkout 335a0aa9a6e57fef17d0865b2f404e0ca785124b it is catching pokemon as notmal.  So something after that is causing this.

**Comment by incenx** [comment]
@josephmcox but on #335a0aa , it has teleportation problem...

**Comment by josephmcox** [comment]
Ahh didnt know that. I have been using it and not getting banned.... its been working great

**Comment by leadboots5** [comment]
broken as of https://github.com/PokemonGoF/PokemonGo-Bot/pull/735 to help narrow it down to 7 or so builds

**Comment by bstpierr** [comment]
I tried checking out 335a0aa and got softbanned after a few minutes due to teleporting between stops.

I've switched to using the master branch until this is resolved.

**Comment by josephmcox** [comment]
Does master branch work for catching at a decent speed?

**Comment by bstpierr** [comment]
It still seems to not see as many mons as I would expect, but at least it finds something between stops once in a while.

In my current session it caught 5 Pokemons in the last 10 minutes, between 8 Pokestops.... and it caught one more as I wrote this, so maybe it's not so bad after all :)

**Comment by Foulwerp** [comment]
Master works correctly, catching is only broken on dev.

**Comment by bstpierr** [comment]
Master does seem to have some strange pathing between stops though. I'm not sure why that southernmost stop was so appealing.

<img width="495" alt="screen shot 2016-07-25 at 3 19 47 pm" src="https://cloud.githubusercontent.com/assets/464763/17114066/5a653dca-527b-11e6-9b1c-e12c7e9cf1f0.png">

**Comment by josephmcox** [comment]
How did you get that map?

**Comment by rguedes** [comment]
@josephmcox run this command on root folder: `git submodule foreach git pull origin master`

**Comment by 03t02** [comment]
@josephmcox https://github.com/OpenPoGo/OpenPoGoWeb

**Comment by IcicleTF** [comment]
Yeah, same issue. Even when setting it to "poke" mode instead of "all" or "farm", it will catch a pokemon or two and then perpetually go between certain Pokestops.

**Comment by bstpierr** [comment]
Is anyone looking into this? I feel as though all development is blocked on the dev branch until this can be resolved.

**Comment by dddbliss** [comment]
Issue appears to be part of the Stepper refactor. Appears that map cells are not refreshed, so new catchable pokemon are not seen. I am fiddling with it, but python is new to me.

Revision 586a0dc

**Comment by bstpierr** [comment]
I think the biggest issue is the fort loop here: https://github.com/PokemonGoF/PokemonGo-Bot/blob/dev/pokemongo_bot/__init__.py#L206

This won't break out until all the forts in a cell have been visited, or the bot encounters a stop it couldn't spin.

This is wrong for a few reasons. Most obviously, it prevents the PokemonCatchWorker from acting for a long time. It also means that the bot will artificially move with a cell more than it should. The nearest stop may actually be closer, but in a neighboring cell. This second point may be the root cause of https://github.com/PokemonGoF/PokemonGo-Bot/issues/830 Furthermore, this loop is essentially another navigator that hijacked the navigation from SpiralNavigator.

I believe this is a symptom of the current design (or at the very least, the partial refactor of the new design). Are workers expected to call other workers during their work, or should they periodically pause their work to allow others to act? Without a well-defined protocol between the two, bugs like this will continue to appear.

**Comment by krypttr** [comment]
Still not resolved. Bot roams between forts and no pokemon in sight.

**Comment by bstpierr** [comment]
The PR hasn't been merged. It's not expected to be working until it is.

That's why I cherry-picked it into my own branch in the meantime :)

**Comment by rlking** [comment]
git fetch https://github.com/binarydepartment/PokemonGo-Bot.git bugfix/catch-pokemon
git cherry-pick 79165fad5bff7ae29a229283fdaaa23d217e2c27

**Comment by krypttr** [comment]
Thank you @rlking. But, I'm pretty new to git and have no idea on how to "resolve the conflicts". However, I'll try.

**Comment by TheGoldenXY** [comment]
Please upgrade to the latest `dev` branch by running this command:
```
git reset --hard
git pull -f
```

Then run this command:
`pip install --upgrade -r requirements.txt`

If issue persists, open a new issue.

/closed
